### ISSUE TITLE
Character positions in PDF output are rounded to int, causing inconsistent spacing

### ISSUE THREAD

**Issue by DmitryFromDubna** [wkhtmltopdf/wkhtmltopdf#3260]
Hello,
I can see the issue on Windows and CentOS 6.7, but not on Ubuntu 14.04. See attached screnshots:
A test file converted on Ubuntu vs same file converted on CentOS:
![wkhtmltopdf ubuntu-vs-centos](https://cloud.githubusercontent.com/assets/24712324/21418724/3ca878d6-c834-11e6-9a02-f192dc497de1.png)
Note that Ubuntu file has floating point coordinates, while Centos has all integers.
![wkhtmltopdf ubuntu-vs-centos rounding](https://cloud.githubusercontent.com/assets/24712324/21418725/3ca8e1a4-c834-11e6-9648-e60cf7e12fc0.png)
This is the test file:
[Testme.html.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/668244/Testme.html.zip)

Both systems had msttcorefonts installed for a closer comparison, but obviously it is not specific to particular fonts.

I am getting pretty much the same result on Windows, although with different scaling; I only attach Ubuntu and CentOS because they are more apples-to-apples.

Tested using wkhtmltopdf 0.12.4.

**Comment by DmitryFromDubna** [comment]
I suspect it is Qt issue actually, but I wonder if it is known? It should be affecting everyone, but I don't see any relevant reports.

**Comment by ashkulz** [comment]
Was the same binary used? Then it is really strange and definitely a bug in Qt.

**Comment by DmitryFromDubna** [comment]
Yes, same binary from wkhtmltox-0.12.4_linux-generic-amd64.tar.xz

**Comment by DmitryFromDubna** [comment]
Just in case, here are the PDFs:
[Testme.wkhtmltopdf.opt.centos.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/668461/Testme.wkhtmltopdf.opt.centos.pdf)
[Testme.wkhtmltopdf.opt.ubuntu.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/668460/Testme.wkhtmltopdf.opt.ubuntu.pdf)
I used this line to produce them, on both platforms:
wkhtmltopdf --disable-javascript --disable-local-file-access --page-height 279.39999999999998 --page-width 215.89999999999998 -L 12.699999999999999 -T 12.699999999999999 -R 12.699999999999999 -B 12.699999999999999 Testme.html Testme.wkhtmltopdf.opt.pdf

**Comment by mikerockett** [comment]
Yeah, definitely noticed kerning issues on Windows. With some fonts, it doesn't appear to be a problem. With others, it is. Have not yet tried on centos, especially considering that my production server is cloudlinux.

**Comment by crystalfp** [comment]
On Windows, if the `<body>` contains no `font-family` CSS, then the spacing between letters is horribly messed up. Instead if I add `body {font-family: Arial,san-serif;}` CSCS rule, it renders perfectly. Instead of Arial Calibri is also OK. It works also with serif fonts. I tried just with Garamond.
Hope it helps.
mario

**Comment by DmitryFromDubna** [comment]
Hi @crystalfp, this does not seem to work for me. Would you mind correcting my test file [Testme.html.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/668244/Testme.html.zip) and attaching here?

**Comment by DmitryFromDubna** [comment]
Additional info: same error exists in conversion to SVG, but conversion to PNG works perfectly (tested on RHEL): 
![comparepdfsvgpng](https://cloud.githubusercontent.com/assets/24712324/21484495/c716e566-cba4-11e6-803a-a91b37a93720.png)
Attached SVG and PNG outputs in [Testme.png.svg.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/673173/Testme.png.svg.zip)

**Comment by crystalfp** [comment]
It happens also with your test file.
[Testme.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/673539/Testme.pdf)
Thanks for looking at it!
mario

**Comment by crystalfp** [comment]
Win 7, wkhtmltopdf version: Windows (MSVC) 0.12.4  64-bit 	

It happens also with your test file.
[Testme.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/673539/Testme.pdf)
Thanks for looking at it!
mario

**Comment by r0polach** [comment]
Win10,  wkhtmltopdf (MinGW) 0.12.4 64-bit:

With no font-family CSS, the result is horrible.
But also with specifying Arial, Calibri or Garamond,
the result is very bad. And Garamond is even worse,
when using accented characters.

**Comment by GitteHubert** [comment]
Concerning the font kerning problems there's Issue #1527 in which a workaround was suggested:
Convert TTF to SVG and then to WOFF. Use WOFF in your documents.

Maybe this issue and #1527 share the same root cause.

**Comment by Tomsgu** [comment]
Does this still happens in 0.12.5 version?
